replication-api-impl/src/main/java/com/connexta/replication/api/impl/worker/WorkerManager.java

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.replication.api.impl.worker;
+
+import com.connexta.replication.api.data.NotFoundException;
+import com.connexta.replication.api.data.ReplicationPersistenceException;
+import com.connexta.replication.api.data.Site;
+import com.connexta.replication.api.impl.NodeAdapters;
+import com.connexta.replication.api.persistence.SiteManager;
+import com.connexta.replication.api.queue.QueueBroker;
+import com.connexta.replication.spring.ReplicationProperties;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Class responsible for managing item workers. It will monitor sites currently available in storage
+ * and create/destroy item workers accordingly for each site, respecting each site's parallelism
+ * factor.
+ */
+public class WorkerManager {
```

Comment:
❓ Should this bean have a destroy() method such that spring can shut it down? This is where you were turn around and shutdown all pools.